eo-maven-plugin/src/main/java/org/eolang/maven/SafeMojo.java

```diff
@@ -285,4 +299,33 @@ private static void closeTojos(final Closeable res) throws MojoFailureException
         }
     }
 
+    /**
+     * Interrupts thread if time limit is reached.
+     *
+     * @param thread Thread to interrupt.
+     * @param sec Time limit in seconds.
+     */
+    private static void waitAndInterrupt(final Thread thread, final int sec) {
+        Executors.newSingleThreadExecutor().submit(
```

Comment:
@volodya-lombrozo why making an executor for a single thread? how it's better than just `new Thread`?